atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2576,6 +2584,36 @@ private Multimap<TableReference, Cell> getCellsToScrubByTable(State expectedStat
         return tableRefToCells;
     }
 
+    @Override
+    public long getAgeMillis() {
+        return System.currentTimeMillis() - timeCreated;
+    }
+
+    @Override
+    public TransactionReadInfo getReadInfo() {
+        return keyValueService.getOverallReadInfo();
+    }
+
+    @Override
+    public void reportExpectationsCollectedData() {
+        if (!isDefinitivelyCommitted() && !isAborted()) {
```

Comment:
While correct, I think this would be easier to read 9by applying De Morgen laws) -> !(isDefinitivelyCommitted || isAborted)